src/http/client.cr

```diff
@@ -807,48 +799,39 @@ class HTTP::Client
     end
   end
 
-  {% if flag?(:without_openssl) %}
-    protected def self.tls_flag(uri, context : Nil)
-      scheme = uri.scheme
+  protected def self.tls_flag(uri, context : TLS) : TLS
+    scheme = uri.scheme
+    raise ArgumentError.new("Missing scheme: #{uri}") if !scheme
+    {% begin %}
+      {% if flag?(:without_openssl) %}
       case scheme
-      when nil
-        raise ArgumentError.new("Missing scheme: #{uri}")
-      when "http"
-        false
-      when "https"
-        true
-      else
-        raise ArgumentError.new "Unsupported scheme: #{scheme}"
-      end
-    end
-  {% else %}
-    protected def self.tls_flag(uri, context : OpenSSL::SSL::Context::Client?)
-      scheme = uri.scheme
+      when "http"  then false
+      when "https" then true
```

Comment:
These lines are missing one indent.

Reply:
Nope, they're fine (formatter agrees).

Reply:
Fixed with the simplification of the macro, I guess.